app/tiebout.hoon

```diff
@@ -0,0 +1,269 @@
+  ^-  (quip move _this)
+  (poke-tiebout-action act)
+::
+::  +poke-tiebout-action: main action handler
+::
+++  poke-tiebout-action
+  |=  act=action
+  ^-  (quip move _this)
+  ?-  -.act
+    $king        (set-king +.act)
+    $token       (set-token +.act)
+    $baseurl     (set-baseurl +.act)
+    $add-circle  (add-circle +.act)
+    $del-circle  (del-circle +.act)
+    $notify      (send-notify +.act)
+  ==
+::
+::  +add-circle: add circle and subscribe for updates
+::
+++  add-circle
+  |=  nom=name:hall
+  ^-  (quip move _this)
+  :_  this(circles.sta (~(put by circles.sta) nom 0))
+  [ost.bol %peer /our/[nom] [our.bol %hall] /circle/[nom]/config/grams]~
+::
+::  +del-circle: delete circle and unsubscribe from updates
+::
+++  del-circle
+  |=  nom=name:hall
+  ^-  (quip move _this)
+  :_  this(circles.sta (~(del by circles.sta) nom))
+  [ost.bol %pull /our/[nom] [our.bol %hall] ~]~
+::
+::  +set-king: set king @p
+::
+++  set-king
+  |=  kng=@p
+  ^-  (quip move _this)
+  [~ this(king.sta kng)]
+::
+::  +set-token: set iOS device token @t
+::
+++  set-token
+  |=  tok=@t
+  ^-  (quip move _this)
+  [~ this(token.sta tok)]
+::
+::  +set-baseurl: set base url @t
+::
+++  set-baseurl
+  |=  burl=@t
+  ^-  (quip move _this)
+  [~ this(baseurl.sta burl)]
+::
+::  +send-notify: if king, send hiss. if not, do nothing.
+::
+++  send-notify
+  |=  not=notification
+  ^-  (quip move _this)
+  ?:  =(king.sta our.bol)
+    :_  this
+    [ost.bol %hiss /request [~ ~] %httr %hiss (create-apns-request not)]~
+  [~ this]
+::
+::  +diff-hall-prize: receive new circle data
+::
+++  diff-hall-prize
+  |=  [wir=wire piz=prize:hall]
+  ^-  (quip move _this)
+  ?+  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ::
+  ::  %our: set config of circle and iterate through messages, sending
+  ::  notifications for all messages where number is higher than our last-read
+  ::
+      {%our @ @}
+    ?>  ?=(%circle -.piz)
+    =/  nom/name:hall  i.t.wir
+    =/  red/@ud  red.loc.cos.piz
+    [~ this(circles.sta (~(put by circles.sta) nom red))]
+  ==
+::
+::  +reap: recieve acknowledgement for peer
+::
+++  reap
+  |=  [wir=wire err=(unit tang)]
+  ^-  (quip move _this)
+  ?~  err
+    [~ this]
+  ?+  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+    ::
+      {%our @ @}
+    ?<  ?=(~ t.wir)
+    [~ this]
+  ==
+::
+::  +quit: receive subscription failed, resubscribe
+::
+++  quit
+  |=  wir=wire
+  ^-  (quip move _this)
+  ?+  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+    ::
+      {%our @ @}
+    ?<  ?=(~ t.wir)
+    :_  this
+    =/  doc/dock  [our.bol %hall]
+    [ost.bol %peer /our/[i.t.wir] doc /circle/[i.t.wir]/config/grams]~
+  ==
+
+::
+::  +diff-hall-rumor: receive message or a read event from a hall circle
+::
+++  diff-hall-rumor
+  |=  [wir=wire rum=rumor:hall]
+  ^-  (quip move _this)
+  ?+  wir
+    (mean [leaf+"invalid wire for diff: {(spud wir)}"]~)
+  ::
+  ::  %our
+  ::
+      {%our @ @}
```

Comment:
While these are technically correct and will successfully match `/our/whatever` aka `[%our %whatever ~]`, it's better to describe a limited-length wire by actually ending with `~` in the type description.